eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1742,6 +1742,21 @@ protected function copyTranslationsFromPublishedVersion(APIVersionInfo $versionI
         $this->internalUpdateContent($versionInfo, $updateStruct);
     }
 
+    protected function isHashEqual($fieldType, $newValue, $fieldValue): bool
+    {
+        $newHash = $fieldType->toHash($newValue);
+        $currentHash = $fieldType->toHash($fieldValue);
+        if ($newHash === $currentHash) {
```

Comment:
if `Image` is specific to compare, should it not be in the first `if`?

Reply:
> if `Image` is specific case to compare, should it not be in the first `if`?

Is there a case where for Image this operation would give false positive? If not, then it's fine as-is, because the latter elseif block condition would occur for less cases (there are N field types, out of which Image is just one).
BUT...

Reply:
>Is there a case where for Image this operation would give false positive?

Well I do not know, but I think that way we are more robust against future problems.